test/functional/logforwarding/message_format_test.go

```diff
@@ -0,0 +1,52 @@
+package logforwarding
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers"
+)
+
+var _ = Describe("[LogForwarding] Functional tests for message format", func() {
+
+	const (
+		timeStamp = "2020-11-04T18:13:59.061892+00:00"
+		message  = "Functional test message"
+		indexName = "app-write"
+	)
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should parse application log format correctly", func() {
+		// Write log line as input to fluentd
+		applicationLogLine := fmt.Sprintf("%s stdout F %s $n", timeStamp, message)
+		Expect(framework.WritesMessageToApplicationLogs(applicationLogLine,1)).To(BeNil())
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom("fluentforward")
+
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+		logs, err := helpers.ParseLogs(raw)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+
+		Expect(logs[0].Timestamp).Should(Equal(timeStamp), fmt.Sprintf("Expected correct Timestamp in %#v", logs[0]))
```

Comment:
I think we should reconcider how we verify.  The intent is to verify the output format.  The result could have missing fields we expect or extra we do not.  Maybe we should deserialize into generic map and compare against that.  Or maybe we should be doing a raw string comparison if we can ensure the output is repeatable

Reply:
@jcantrill I created code that uses template on the fields to compare input and output ... the only thing that you need to do now as to test is either (1) specify concreate values to test against or (2) start a string with pre so it will check for pre-fix or (3) write '*' in a field and than we test that there is 'something' in the output field ... 

This seems very flexible + the code for the test is really short now ... just one line that tests against the input with template fields.